### ISSUE TITLE
RepeatMasker 4.1.6 update

### ISSUE THREAD

**Issue by KristinaGagalova** [bioconda/bioconda-recipes#45328]
Hi, 

Since there are significant updates in the new version, could you please create the package for RepeatMasker 4.1.6?

Thank you in advance

**Comment by TobyBaril** [comment]
This is not as simple as many will think, due to version 4.1.6 requiring at least a several GB database to be configured, which cannot be stored and shared in github/conda packages which are designed for sharing code only. 

I have submitted a potential workaround for this: #45513 , but will need to see if this works with conda rules.

**Comment by KristinaGagalova** [comment]
Hi @TobyBaril 
I have a suggestion.
Instead of downloading the database and configuring it, why don't we try to change manually the configuration file for a placeholder? The user would then pass the ```-libdir``` parameter later. This is how I bypassed the db download for my docker container

https://github.com/KristinaGagalova/pante2/blob/ae44d6c5955f3be42e91811c4610e070eab5605e/containers/pante2_env.Dockerfile#L109-L132

**Comment by TobyBaril** [comment]
Yes great idea! This worked for version 4.1.5 but the Perl configure script for 4.1.6 requires the actual file to correctly configure. I tried with a placeholder and it just exits and asks me to download the correct file even with the same name etc…it has a minimum requirement of the 0 partition of the database for correct configuration.

**Comment by TobyBaril** [comment]
I'm going to try and see if it will configure properly with a single entry from the small database, this seems like it is worth trying...

**Comment by ptrebert** [comment]
Just to chime in here: if at all possible, it would really --- and I mean really, really --- be great if there was a way to make the RepeatMasker installation via Conda more straightforward on offline systems such as HPC infrastructure.

I am aware that certain Conda packages download annotations/DBs during --- or after --- installation from some other source (= not conda channels). This will always break / never work on offline systems. So far, the way around that for us was to use `conda-pack / -unpack`, which worked fine. But `conda-pack` does not work for RepeatMasker. Given the non-trivial dependency resolution and configuration (see, e.g. here #9988 ), I am not surprised that after unpacking, RepeatMasker just quits with a `bad interpreter` error, so some paths are likely not properly reconfigured by `conda-unpack` after migrating the environment.

**Comment by KristinaGagalova** [comment]
Since this looks so painful, I reiterate my previous message: we don't need to download databases; we can change the config file manually. That's the fastest way and will hopefully work out for the checkups. 
Please let me know if you want to give it another stub, or I can try to modify the configuration through scripting. 

Thank you in advance

**Comment by mikecuoco** [comment]
Hi all, I just wanted to check-in on this issue. Has this been resolved?

**Comment by KristinaGagalova** [comment]
Hi, 
The issue has not been resolved yet, the PR is still open.

**Comment by mikecuoco** [comment]
@TobyBaril did you have any luck with this solution?

> I'm going to try and see if it will configure properly with a single entry from the small database, this seems like it is worth trying...

**Comment by TobyBaril** [comment]
Hi @mikecuoco, no luck with configuring with a single entry. I've had a good chat with the authors and they acknowledge the issues with the database configuration in Dfam3.8 and the new RepeatMasker. 

RepeatMasker 4.1.6 with the root partition has been force-pushed to bioconda though, see: https://github.com/bioconda/bioconda-recipes/pull/45513

**Comment by mikecuoco** [comment]
@TobyBaril I noticed that force push, that makes sense now. Perhaps we can force a merge to master despite the CI failures.

**Comment by mikecuoco** [comment]
@TobyBaril have you tested your personal conda version on OSX? I noticed @abretaud found an issue with the build process there- perhaps we could incorporate that in your PR as well

**Comment by TobyBaril** [comment]
I haven't tested on OSX - I currently don't have a system with enough storage. I guess it is just a case of swapping a couple of bits to make it behave though?

**Comment by mikecuoco** [comment]
let me try quickly
